pkg/microservice/aslan/core/code/service/repo.go

```diff
@@ -87,26 +84,23 @@ func ListRepoInfos(infos []*GitRepoInfo, param string, log *zap.SugaredLogger) (
 
 		}(info)
 
-		//bt 代表branch and tag
-		if param == "" || param == "bt" {
-			wg.Add(1)
-			go func(info *GitRepoInfo) {
-				defer func() {
-					wg.Done()
-				}()
-				projectName := info.Repo
-				if info.Source == CodeHostCodeHub {
-					projectName = info.RepoID
-				}
-				info.Tags, err = CodeHostListTags(info.CodehostID, projectName, strings.Replace(info.Owner, "%2F", "/", -1), "", 0, 0, log)
-				if err != nil {
-					errList = multierror.Append(errList, err)
-					info.ErrorMsg = err.Error()
-					info.Tags = []*Tag{}
-					return
-				}
-			}(info)
-		}
+		wg.Add(1)
+		go func(info *GitRepoInfo) {
+			defer func() {
+				wg.Done()
+			}()
+			projectName := info.Repo
+			if info.Source == CodeHostCodeHub {
+				projectName = info.RepoID
+			}
+			info.Tags, err = CodeHostListTags(info.CodehostID, projectName, strings.Replace(info.Owner, "%2F", "/", -1), "", 0, 0, log)
+			if err != nil {
+				errList = multierror.Append(errList, err)
+				info.ErrorMsg = err.Error()
+				info.Tags = []*Tag{}
+				return
+			}
+		}(info)
```

Comment:
consider write a new function to get both branch and tag info? The client of codehosts will be inited only once, and we can use a option to determine weather to list bran or tag or both.

Reply:
ok refactor